src/main/java/org/broadinstitute/hellbender/tools/walkers/realignmentfilter/RealignmentEngine.java

```diff
@@ -132,8 +132,11 @@ private final static boolean mightSupportInsertion(final CigarElement cigarEleme
         return new ArrayList<>(commonAlignments);
     }
 
+    // note the conversion from exclusive end (BWAMemAlignment) to inclusive end (SimpleInterval) (subtract 1 from the end)
+    // combined with the conversion from 0-based to 1-based contigs (add 1 to start and end)
+    // which has the net result of adding 1 to the start and leaving the end alone
     private static SimpleInterval convertToInterval(final BwaMemAlignment alignment) {
-        return new SimpleInterval(Integer.toString(alignment.getRefId()), alignment.getRefStart(), alignment.getRefEnd());
+        return new SimpleInterval(Integer.toString(alignment.getRefId()), alignment.getRefStart() + 1, alignment.getRefEnd());
```

Comment:
@davidbenjamin Are we sure that the contig is correct?  I think alignment.getRefId() is likely to be 0 indexed, which means that you'll start with contig 0 when you want contig 1.  And obviously it doesn't work for non-numbered contigs.  I think what you need to be doing here is looking it up from bwaMemIndex.getReferenceContigNames().

Reply:
Oh.  I see, it's using the same method to generate the overlap detectors, so that's ok.